bonsai-core/src/main/scala/com/stripe/bonsai/FullBinaryTree.scala

```diff
@@ -59,14 +64,13 @@ object FullBinaryTree {
    *
    * @param tree the tree whose structure we are copying
    */
-  def apply[T](tree: T)(implicit ev: FullBinaryTreeOps.WithLayout[T]): FullBinaryTree[ev.treeOps.BranchLabel, ev.treeOps.LeafLabel] = {
+  def apply[T, B, L](tree: T)(implicit ev: FullBinaryTreeOps[T, B, L]): FullBinaryTree[B, L] = {
```

Comment:
You'll want to require a `B: Layout` and `L: Layout` here. I should really get rid of the default implicits for this so it would just fail to compile, rather than de-optimize silently!